service/src/main/java/uk/nhs/adaptors/gp2gp/common/task/TaskConsumer.java

```diff
@@ -0,0 +1,36 @@
+package uk.nhs.adaptors.gp2gp.common.task;
+
+import java.io.IOException;
+
+import javax.jms.JMSException;
+import javax.jms.Message;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.stereotype.Component;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.common.exception.TaskHandlerException;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class TaskConsumer {
+
+    private final TaskHandler taskService;
+
+    @JmsListener(destination = "${gp2gp.amqp.taskQueueName}")
+    public void receive(Message message) throws IOException, JMSException, TaskHandlerException {
+        var messageID = message.getJMSMessageID();
+        LOGGER.info("Received message from taskQueue {}", messageID);
+        try {
+            taskService.handle(message);
+            message.acknowledge();
+            LOGGER.info("Acknowledged message {}", messageID);
+        } catch (Exception e) {
```

Comment:
Catching all exceptions does not seem to be a good practice. Could we only catch specific ones?